app/controllers/assessments_controller.rb

```diff
@@ -68,10 +68,13 @@ def new
     @threshold_hash = jurisdiction.hierarchical_symptomatic_condition.threshold_condition_hash
     @translations = @assessment.translations
     @contact_info = jurisdiction.contact_info
-    @lang = permitted_params[:lang] if %w[en es es-PR so fr].include?(params[:lang])
-    @lang = 'en' if @lang.nil? # Default to english
+
+    @lang = LegacyLanguages.legacy_language_code?(permitted_params[:lang]) ?
+            LegacyLanguages.translate_legacy_language_code(permitted_params[:lang]) : permitted_params[:lang]
```

Comment:
```suggestion
    @lang = LegacyLanguages.translate_legacy_language_code(permitted_params[:lang]) || permitted_params[:lang]
```

`translate_legacy_language_code` will return `nil` if it's not in `LEGACY_LANGUAGE_MAPPING`
https://github.com/SaraAlert/SaraAlert/blob/c27bc90442b3cc9ae3881ba014e73aa0fc17e9f8/app/lib/legacy_languages.rb#L20-L22

So there is no need for the call to `legacy_language_code?`:
https://github.com/SaraAlert/SaraAlert/blob/c27bc90442b3cc9ae3881ba014e73aa0fc17e9f8/app/lib/legacy_languages.rb#L15-L17

So as far as I can tell, the `legacy_language_code?` method can be removed.

Reply:
I've removed it from this bit of logic, but I kinda like the idea of keeping it. We might want a future method for just determining if it is a legacy language. Thoughts?